src/vanilla/Model/CompositeTypeRb.cs

```diff
@@ -113,6 +114,18 @@ public virtual string ConstructModelMapper()
             return builder.ToString();
         }
 
+        public bool IsResourceGroupRequired()
+        {
+            Regex resourceRegEx = new Regex(@"^(RESOURCE)$", RegexOptions.IgnoreCase);

```

Comment:
I wonder if some of this could go into the runtime, seem like including logic to detect a resource (in order to add this method) in each SDK is not ideal.

Reply:
Such a small change. I do not think it warrants its place in runtime